pkg/awsconfig/awsconfig.go

```diff
@@ -42,9 +42,10 @@ type CredentialsProvider struct {
 }
 
 // NewSharedCredentials helper to create the credentials provider
-func NewSharedCredentials(profile string) *CredentialsProvider {
```

Comment:
@sledigabel, I believe saml2aws supports `AWS_SHARED_CREDENTIALS_FILE` pretty well. I've been setting that before invoking `saml2aws`, but having this built into the app would be pretty handy. Perhaps it would make sense to support it in `locateFileConfig()` (instead of here? not sure) so that it interacts well with the way that env var is used?

Reply:
You're right, it does support it and `saml2aws` already supports it in the said function :-)
https://github.com/Versent/saml2aws/blob/master/pkg/awsconfig/awsconfig.go#L174

As mentioned in the PR, the AWS SDK will also use that same env variable, and thus `saml2aws` and any aws sdk tool like `awscli` would point to the same file. It would not be a problem if the sdk would handle expired credentials, but they do not. The credential process (in PR #595) fixes this problem, however the credentials take precedence over the credential process.

I could work around the issue by having a wrapper that sets `AWS_SHARED_CREDENTIALS_FILE` for `saml2aws` but that makes it harder to distribute vs an additional flag set once in the `configure` process.

Another use case is the use of multiple `IDP account`, they would conflict between themselves (writing into the same file) and the feature would allow to distinguish credentials from different accounts.

Reply:
Re-read your comment @oshaughnessy, I think I understand what you mean now.
`locateFileConfig()` will still be instanciated, and will honour the `AWS_SHARED_CREDENTIALS_FILE` as before if a specific credentials file is not set up (the default mode). That allows the ini file, and the CLI option and the specific SAML2AWS env variable to superseed the `AWS_SHARED_CREDENTIALS_FILE` if they are present.

Thank you for pointing this out, I'll add this to the documentation.

Reply:
Yeah, thanks for clarifying. I have been using a wrapper to set `AWS_SHARED_CREDENTIALS_FILES`, but I don't like it for the same reasons you mentioned. This would get me one step closer to only needing to distribute a configuration script.